python_modules/dagster/dagster_tests/definitions_tests/decorators_tests/test_asset_decorator_with_check_specs.py

```diff
@@ -264,10 +228,27 @@ def asset1():
             ...
 
 
+def test_check_wrong_asset():
+    with pytest.raises(
+        DagsterInvalidDefinitionError,
+        match=re.escape("Invalid asset key"),
+    ):
+
+        @asset(
+            check_specs=[
+                AssetCheckSpec("check1", asset="other_asset", description="desc1"),
+            ]
+        )
+        def asset1():
+            ...
+
+
 def test_multi_asset_with_check():
     @multi_asset(
         outs={"one": AssetOut("asset1"), "two": AssetOut("asset2")},
-        check_specs=[AssetCheckSpec("check1", asset="asset1", description="desc")],
+        check_specs=[
+            AssetCheckSpec("check1", asset=AssetKey(["asset1", "one"]), description="desc")
```

Comment:
all these multi asset test cases had the checks pointing at a non existent asset, since `outs={"one": AssetOut("asset1")` actually creates`AssetKey(["asset1", "one"])`

Reply:
(I was very surprised that `asset1` in `AssetOut("asset1")` becomes the key prefix, not the key)

Reply:
@schrockn more fodder against prefixes